### ISSUE TITLE
concurrency issue: Internal Error occured.

### ISSUE THREAD

**Issue by mfulton26** [spekframework/spek#93]
It doesn't do it every time but running the following spek sometimes causes the below NPE.

``` kotlin
class ParallelSpek : Spek({
    given("some parallel operations") {
        // uses `parallelStream` from `kotlinx-support-jdk8-0.1`
        listOf(1, 2, 3, 4, 5).parallelStream().forEach { int ->
            on("execution") {
                it("should something") {}
                it("should something else") { fail() }
                it("should yet something else") { check(false) }
            }
        }
    }
})
```

```
Internal Error occured.
java.lang.NullPointerException
    at java.util.LinkedList$ListItr.next(LinkedList.java:893)
    at org.jetbrains.spek.junit.JUnitDescriptionCache.create(JUnitDescriptionCache.kt:45)
    at org.jetbrains.spek.junit.JUnitDescriptionCache.get(JUnitDescriptionCache.kt:20)
    at org.jetbrains.spek.junit.JUnitDescriptionCache.create(JUnitDescriptionCache.kt:30)
    at org.jetbrains.spek.junit.JUnitDescriptionCache.get(JUnitDescriptionCache.kt:20)
    at org.jetbrains.spek.junit.JUnitSpekRunner.getDescription(JUnitSpekRunner.kt:23)
    at com.intellij.junit4.JUnit4IdeaTestRunner.getDescription(JUnit4IdeaTestRunner.java:123)
    at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:99)
    at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:42)
    at com.intellij.rt.execution.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:234)
    at com.intellij.rt.execution.junit.JUnitStarter.main(JUnitStarter.java:74)
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke(Method.java:497)
    at com.intellij.rt.execution.application.AppMain.main(AppMain.java:144)
```

**Comment by mfulton26** [comment]
In addition, if the Internal Error does not occur sometimes all of the `on` blocks do not execute.

I'm loving what I see so far with Spek but I do need to be able to run some of the execution for testing in parallel and I would like to avoid synchronizing `DescribeBody` method calls.

Thanks.

**Comment by raniejade** [comment]
What's the use case for this? Spek going some significant changes before `1.0`, this may or may not be applicable when those changes are applied.

**Comment by artem-zinnatullin** [comment]
Can't you move multithreading to `it("something")` so it'll live inside of the test?

Also, I'm not sure that JUnit runner will wait for such tests without explicit blocking inside of the tests.

**Comment by mfulton26** [comment]
If I move the multi-threading to `it("something")` then I will have multiple asserts in a single test.

`JUnitSpekRunner` doesn't have, from what I can tell, any kind of timeout on describing the test. The time for the tests is only the time spent asserting things and not the total execution time as the tests don't actually start until after all of the HTTP requests are finished. This is typical though in time statistics for integration tests.

Basically, I want to do a whole bunch of work in parallel and schedule assertions while I go and then execute all of the assertions at the end and report.

**Comment by raniejade** [comment]
I guess the problem here is Spek not supporting parallel executions of tests (`it`, `on` etc.. blocks), it will definitely happen sometime in the near future.
